poc_mobile_verifier/src/scheduler.rs

```diff
@@ -0,0 +1,254 @@
+use crate::{Error, Result};
+use chrono::{DateTime, Duration, Utc};
+use std::ops::Range;
+
+#[derive(Debug)]
+pub struct Scheduler {
+    pub verification_period_length: Duration,
+    pub reward_period_length: Duration,
+    pub verification_period: Range<DateTime<Utc>>,
+    pub reward_period: Range<DateTime<Utc>>,
+}
+
+impl Scheduler {
+    pub fn new(
+        verification_period_length: Duration,
+        reward_period_length: Duration,
+        last_verified_end_time: DateTime<Utc>,
+        last_rewarded_end_time: DateTime<Utc>,
+        next_rewarded_end_time: DateTime<Utc>,
+    ) -> Self {
+        let verification_period = last_verified_end_time
+            ..((last_verified_end_time + verification_period_length).min(next_rewarded_end_time));
+
+        Self {
+            verification_period_length,
+            reward_period_length,
+            verification_period,
+            reward_period: last_rewarded_end_time..next_rewarded_end_time,
+        }
+    }
+
+    pub fn should_verify(&self, now: DateTime<Utc>) -> bool {
+        now >= self.verification_period.end
+    }
+
+    pub fn should_reward(&self, now: DateTime<Utc>) -> bool {
+        self.verification_period.end == self.reward_period.end && now >= self.reward_period.end
```

Comment:
I don't believe it will cause problems because we always ensure that the `verification_period.end` is never past the `reward_period.end` with this code
```
        let verification_period = last_verified_end_time
            ..((last_verified_end_time + verification_period_length).min(next_rewarded_end_time));
```